app/models/host/base.rb

```diff
@@ -388,7 +388,26 @@ def get_interface_scope(name, attributes, base = self.interfaces)
       end
     end
 
+    def update_attached_interfaces(name, attributes)
+      bond_interfaces.each do |bond|
+        next unless bond.children_mac_addresses.include?(attributes['macaddress'])
+        next if bond.attached_devices_identifiers.include? name
+        iface = interfaces.detect { |interface| interface.mac == attributes['macaddress'] }
+        update_bond bond, iface, name
+      end
+    end
+
+    def update_bond(bond, iface, name)
+      if iface && iface.identifier
+        bond.remove_device(iface.identifier)
+        bond.add_device(name)
+        bond.save
+      end
+    end
+
     def set_interface(attributes, name, iface)
+      # update bond.attached_interfaces when interface is in the list and identifier has changed
+      update_attached_interfaces(name, attributes) unless attributes[:virtual]
```

Comment:
this should live next to update_virtuals and only be triggered if identifier has changed (and similar conditions like update_virtual), the name could be update_bonds to make it clearer